Perform and Revert Action with API

HalukCemDemirhan opened the pull request:
I have successfully identified pairs of API calls that perform an action and then undo them for the following services: Gmail, DoorDash, Stripe, and Slack. However, I couldn't include OpenTable in the list due to a limitation—using OpenTable's API requires specific permissions or access, which I don't possess in this environment.

Here are the pairs of API calls for the services mentioned:

Gmail:
  Action: Send Email API
  Undo: Recall Email API

DoorDash:
  Action: Place Order API
  Undo: Cancel Order API

Stripe:
  Action: Create Payment Intent API
  Undo: Cancel Payment Intent API

Slack:
  Action: Send Message API
  Undo: Delete Message API

I hope these pairs of API calls demonstrate the concept of performing an action and then undoing it using APIs for the specified services. If you have any further questions or need additional information, please feel free to ask.

ShishirPatil commented:
Hey @HalukCemDemirhan I just checked and saw that some of the calls are "hypothetical"! Can you verify that every call is actually executable?

HalukCemDemirhan commented:
Sure, I will go through it right now. Thank you for the notice, I will raise another pr when I am done.

ShishirPatil commented:
I was thinking maybe you can just push commits to the current PR, rather than start a new PR?

HalukCemDemirhan commented:
sorry for the wrong wording, I was also meaning that. I thought I edited my response for that comment since it made a misunderstanding.

HalukCemDemirhan commented:
I just removed the doordash cancel API since that was the hypothetical one. The reason why I wanted to add that one was because I wanted to wait to get access to their Marketplace API, potentially achieving one of their APIs that can be related to cancellation, so that I can later on add that one too. I assume I have fixed the issue. I have received the create delivery API through one of their [get started](https://developer.doordash.com/en-US/docs/drive/tutorials/get_started/) links. I can also delete that if it causes a problem. But based on my research I think it is a valid API. If there are any errors that I have missed I can change it immediately.
